src/table/dtable.jl

```diff
@@ -258,6 +258,27 @@ function length(table::DTable)
     return sum(chunk_lengths(table))
 end
 
+function first(table::DTable, rows::UInt)
+    if nrow(table) == 0
+        return table
+    end
+
+    chunk_length = chunk_lengths(table)[1]
```

Comment:
chunk lengths are not guarenteed to be equal
some may even be empty

Reply:
Hi @krynju. If this is the case, is there any way to retrieve the original `chunksize`? If I'm not wrong it's not stored as a property of `DTable`s.

On another note: suppose for a `DTable` I have `chunksize` greater than number of rows in the table. In that case, won't I lose information about what `chunksize` I passed?

Reply:
Yeah, I think it was an early design decision to make chunksize an argument of the constructor for the initial partitioning and later ignore it (and for that reason not store it either)

I think including the original chunksize in the logic would also be a bit confusing and would make it more complex, but if we have any use case for that then we can revisit this 

I did think of caching the current chunk sizes, because generally that information doesn't change in a dtable (after you manipulate a dtable it becomes a new dtable)
We already cache the schema so a similar mechanism could be used

Reply:
and for this you can just use chunk_legths as you did 
https://github.com/JuliaParallel/DTables.jl/blob/9fcbe237e0c6ddd6b6f2880f33347efe99a76fdd/src/table/dtable.jl#L252C10-L255

Reply:
> On another note: suppose for a DTable I have chunksize greater than number of rows in the table. In that case, won't I lose information about what chunksize I passed?

You will and you will only get one partition in the dtable

Reply:
> Yeah, I think it was an early design decision to make chunksize an argument of the constructor for the initial partitioning and later ignore it (and for that reason not store it either)
> 
> I think including the original chunksize in the logic would also be a bit confusing and would make it more complex, but if we have any use case for that then we can revisit this
> 
> I did think of caching the current chunk sizes, because generally that information doesn't change in a dtable (after you manipulate a dtable it becomes a new dtable) We already cache the schema so a similar mechanism could be used

@krynju how about this: to get the chunksize, can I get the maximum value from chunk_lengths?

Reply:
Again, not guaranteed. Why do you need the original chunksize?